raiden-ts/src/types.ts

```diff
@@ -49,3 +49,15 @@ export interface RaidenEpicDeps {
   userDepositContract: UserDeposit;
   main?: { signer: Signer; address: Address };
 }
+
+export interface ChangeEvent<T, P> {
+  readonly type: T;
+  readonly payload: P;
+}
+
+export type OnChange<T, P> = (event: ChangeEvent<T, P>) => void;
+
+export enum Deposit {
+  APPROVED,
```

Comment:
May I suggest using string literals for enums? Easier to work on JS and more consistent with the events/actions as well
```suggestion
  APPROVED = 'APPROVED',
```